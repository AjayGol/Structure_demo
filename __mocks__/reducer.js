import { combineReducers } from 'redux';
import AccountReducer from '../src/modules/account/reducer';
import BrokerReducer from '../src/modules/broker/reducer';
import HomeReducer from '../src/modules/home/reducer';
import SearchReducer from '../src/modules/search/reducer';
import WatchlistReducer from '../src/modules/watchlist/reducer';
import OrdersReducer from '../src/modules/watchlist/reducer';
import NotificationReducer from '../src/modules/notification/reducer';
import buyAssetReducer from '../src/modules/buy-assets/reducer';

export default combineReducers({
  account: AccountReducer,
  home: HomeReducer,
  watchlist: WatchlistReducer,
  orders: OrdersReducer,
  search: SearchReducer,
  broker: BrokerReducer,
  buyAsset: buyAssetReducer,
  notification: NotificationReducer,
});
