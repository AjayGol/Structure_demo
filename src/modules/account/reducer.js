import {
  SET_SAFE_AREA_INTENT,
  API_WAITING_TIME,
  START_LOADING,
  SET_USER_DETAIL,
  CUSTOM_POPUP,
  EVENT_DATA,
  SET_USER_OTHER_DETAIL,
} from './constant';

export const initStateAccount = {
  email: process.env.NODE_ENV === 'development' ? 'ketan@gmail.com' : '',
  password: process.env.NODE_ENV === 'development' ? 'test1234' : '',
  userName: '',
  token: '',
  safeAreaInsetsData: { top: 0, bottom: 0, left: 0, right: 0 },
  safeAreaInsetsDataChat: { top: 0, bottom: 0, left: 0, right: 0 },
  safeAreaInsetsDefault: { top: 0, bottom: 0, left: 0, right: 0 },
  apiWaitingTime: 0,
  isLoading: false,
  customPopUp: { isShow: false, data: {} },
  userDetails: {
    id: '',
    name: '',
    username: '',
    mobile: '',
    password: '',
    birthday: '',
    gender: '',
    address: [],
    tradingFrequency: 1,
    brokers: [],
    brokerLoginDetails: {},
    tradingFamiliarity: [],
    traderProgress: {
      traderReturns: 0,
      followerReturns: 0,
      AUM: 0,
      followedPortfolioCount: 0,
      _id: '',
      trader: '',
      id: '',
      twitterUserData: {
        isLoggedIn: false,
      },
    },
    userDetailsOtherData: {
      watchlistsLength: 0,
      ifFirstTradeExecuted: false,
    },
    profileCompletionRatio: {
      profile: 0,
      twitter: 0,
      pan: 0,
      brokers: 0,
      knowledge: 0,
    },
    eventData: false,
  },
};

export default (state = initStateAccount, action) => {
  switch (action.type) {
    case SET_SAFE_AREA_INTENT: {
      return {
        ...state,
        safeAreaInsetsDefault: action.payload,
        safeAreaInsetsData: action.payload,
      };
    }
    case START_LOADING: {
      return {
        ...state,
        isLoading: action.payload,
      };
    }
    case API_WAITING_TIME: {
      return {
        ...state,
        apiWaitingTime: action.payload,
      };
    }
    case SET_USER_DETAIL: {
      return {
        ...state,
        userDetails: action.payload,
        userDetailsOtherData: action.payloadAll,
      };
    }
    case SET_USER_OTHER_DETAIL: {
      return {
        ...state,
        userDetailsOtherData: action.payload,
      };
    }
    case CUSTOM_POPUP: {
      return {
        ...state,
        customPopUp: action.payload,
      };
    }
    case EVENT_DATA: {
      return {
        ...state,
        eventData: action.payload,
      };
    }

    default:
      return state;
  }
};
