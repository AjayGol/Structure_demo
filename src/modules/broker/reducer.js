import { ANGEL_URL } from './constant';

export const initStateBroker = {
  angelURL: '',
};

export default (state = initStateBroker, action) => {
  switch (action.type) {
    case ANGEL_URL: {
      return {
        ...state,
        angelURL: action.payload,
      };
    }

    default:
      return state;
  }
};
