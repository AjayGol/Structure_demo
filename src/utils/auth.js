export const GOOGLE_SIGN_IN_CONFIG = {
  webClientId:
    '500383754555-6249knhcfu3mojs0vgkmnjk4v0tcri9f.apps.googleusercontent.com',
  offlineAccess: true,
};

export const SUPPORT_DIGITS_PLUGIN = false;
