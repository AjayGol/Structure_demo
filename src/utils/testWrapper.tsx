// @ts-ignore
import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../../__mocks__/reducer';
const store = createStore(reducer);

function render(
  ui: React.ReactElement<
    any,
    | string
    | ((
        props: any,
      ) => React.ReactElement<
        any,
        string | any | (new (props: any) => React.Component<any, any, any>)
      > | null)
    | (new (props: any) => React.Component<any, any, any>)
  >,
  { locale = 'en', ...renderOptions } = {},
) {
  const Wrapper: React.FC = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react-native';

// override React Testing Library's render with our own
export { render };
