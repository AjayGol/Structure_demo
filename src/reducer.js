import { combineReducers } from 'redux';
import AccountReducer from './modules/account/reducer';
import HomeReducer from './modules/home/reducer';
import SearchReducer from './modules/search/reducer';
import BrokerReducer from './modules/broker/reducer';
import NotificationReducer from './modules/notification/reducer';
import { initStateAccount } from './modules/account/reducer';
import { initStateHome } from './modules/home/reducer';
import WatchlistReducer from './modules/watchlist/reducer';
import { initStateWatchlist } from './modules/watchlist/reducer';
import { initStateOrders } from './modules/orders/reducer';
import OrdersReducer from './modules/orders/reducer';
import { initStateSearch } from './modules/search/reducer';
import { initStateBroker } from './modules/broker/reducer';
import { initStateNotification } from './modules/notification/reducer';
import { initStateBuyAsset } from './modules/buy-assets/reducer';
import buyAssetReducer from './modules/buy-assets/reducer';

export const appDefaultReducer = {
  account: initStateAccount,
  home: initStateHome,
  orders: initStateOrders,
  search: initStateSearch,
  broker: initStateBroker,
  watchlist: initStateWatchlist,
  buyAsset: initStateBuyAsset,
  notification: initStateNotification,
};

const appReducer = combineReducers({
  account: AccountReducer,
  home: HomeReducer,
  watchlist: WatchlistReducer,
  orders: OrdersReducer,
  search: SearchReducer,
  broker: BrokerReducer,
  buyAsset: buyAssetReducer,
  notification: NotificationReducer,
});

export default function rootReducer(state, action) {
  let finalState = appReducer(state, action);
  if (action.type === 'RESET_STORE') {
    finalState = appDefaultReducer; //resetReducer(finalState, action);
  }
  return finalState;
}
