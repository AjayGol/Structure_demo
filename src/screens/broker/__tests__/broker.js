import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '../../../utils/testWrapper';
import { render as renderLibrary } from '@testing-library/react-native';
import reducer from '../../../../__mocks__/reducer';
import BrokerView from '../broker';
const store = createStore(reducer);

let tree;
let title = 'Testing';
beforeEach(() => {
  tree = render(
    <Provider store={store}>
      <BrokerView />
    </Provider>,
  );
});

test(`${title} - renders correctly`, () => {
  expect(tree).toMatchSnapshot();
});

it(`${title} - loading test`, () => {
  const screen = render(<BrokerView />);
  screen.getByTestId('activity');
});

it(`${title} - not loading test`, () => {
  const screen = render(<BrokerView angelURL={'https://www.google.com'} />);
  screen.getByTestId('WebView');
});
