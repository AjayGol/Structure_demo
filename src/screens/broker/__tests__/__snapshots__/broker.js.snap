// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing - renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E7F4FB",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "elevation": 1,
          "shadowColor": "#164672",
          "shadowOffset": Object {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.3,
        },
        Object {
          "height": 68,
        },
      ]
    }
  >
    <BVLinearGradient
      angle={180}
      angleCenter={
        Object {
          "x": 0.5,
          "y": 0.5,
        }
      }
      colors={
        Array [
          4281175782,
          4280908484,
        ]
      }
      endPoint={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      locations={
        Array [
          0,
          1,
        ]
      }
      startPoint={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 26,
          },
          Object {
            "paddingTop": 0,
          },
          Object {},
        ]
      }
      useAngle={true}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "top": 2,
            },
            Object {
              "marginBottom": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "top": 2,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Viga-Regular",
                  "fontSize": 21,
                  "lineHeight": 25,
                  "textAlign": "center",
                }
              }
            >
               
               
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 15,
                "left": 15,
                "right": 15,
                "top": 15,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "icon_back",
                }
              }
              style={
                Object {
                  "height": 12,
                  "marginRight": 25,
                  "tintColor": "#FFFFFF",
                  "width": 7,
                }
              }
            />
            <Text
              adjustsFontSizeToFit={true}
              minimumFontScale={0.3}
              numberOfLines={1}
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Viga-Regular",
                  "fontSize": 21,
                  "lineHeight": 25,
                  "textAlign": "center",
                }
              }
            >
               
              CONNECT ACCOUNT TO ANGEL BROKING
               
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
    </BVLinearGradient>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#59595C"
      hidesWhenStopped={true}
      size="large"
      testID="activity"
    />
  </View>
</View>
`;
